cpu/esp_common/esp-wifi/esp_wifi_netdev.c

```diff
@@ -903,6 +903,62 @@ static wifi_config_t wifi_config_ap = {
 };
 #endif /* (defined(MCU_ESP8266) && !defined(MODULE_ESP_NOW)) || defined(MODULE_ESP_WIFI_AP) */
 
+
+#if (defined(MCU_ESP32) && defined(MODULE_ESP_WIFI_CSI))
+void __attribute__((weak)) esp_wifi_csi_rx_cb(void *ctx, wifi_csi_info_t *info)
+{
+    if (!info || !info->buf) {
+        ESP_WIFI_LOG_ERROR("ERROR: esp_wifi_csi_rx_cb missing info or buf");
+        return;
+    }
+    printf("{");
+    printf("\"mac\": \"%02x:%02x:%02x:%02x:%02x:%02x\",",
+           (unsigned)info->mac[0], (unsigned)info->mac[1], (unsigned)info->mac[2],
+           (unsigned)info->mac[3], (unsigned)info->mac[4], (unsigned)info->mac[5]);
+    printf("\"rssi\": %d,", info->rx_ctrl.rssi);
+    printf("\"rate\": %d,", info->rx_ctrl.rate);
+    printf("\"sig_mode\": %d,", info->rx_ctrl.sig_mode);
+    printf("\"mcs\": %d,", info->rx_ctrl.mcs);
+    printf("\"cwb\": %d,", info->rx_ctrl.cwb);
+    printf("\"smoothing\": %d,", info->rx_ctrl.smoothing);
+    printf("\"not_sounding\": %d,", info->rx_ctrl.not_sounding);
+    printf("\"aggregation\": %d,", info->rx_ctrl.aggregation);
+    printf("\"stbc\": %d,", info->rx_ctrl.stbc);
+    printf("\"fec_coding\": %d,", info->rx_ctrl.fec_coding);
+    printf("\"sgi\": %d,", info->rx_ctrl.sgi);
+    printf("\"noise_floor\": %d,", info->rx_ctrl.noise_floor);
+    printf("\"ampdu_cnt\": %d,", info->rx_ctrl.ampdu_cnt);
+    printf("\"channel\": %d,", info->rx_ctrl.channel);
+    printf("\"secondary_channel\": %d,", info->rx_ctrl.secondary_channel);
+    printf("\"timestamp\": %d,", info->rx_ctrl.timestamp);
+    printf("\"ant\": %d,", info->rx_ctrl.ant);
+    printf("\"sig_len\": %d,", info->rx_ctrl.sig_len);
+    printf("\"rx_state\": %d,", info->rx_ctrl.rx_state);
+    printf("\"first_word_invalid\": %d,", info->first_word_invalid);
+
+    printf("\"buf\": [");
+    for (int i = 0; i < info->len; i++) {
+        printf("%d", info->buf[i]);
```

Comment:
Why are raw data in a buffer given as 8-bit integers?

Reply:
true, they should probably be unsigned

Reply:
I'm wondering why the data are defined as `int8_t` in `wifi_csi_info_t` :thinking:

Reply:
a bit silly but it is unsigned now